src/robos.py

```
@@ -33,101 +34,179 @@ def __init__(self, id_robo, grid, robots_info, locks):
+            
+            # Checar condição de vitória (apenas 1 robô vivo)
+            with self.locks['robots_mutex']:
+                vivos = [r_id for r_id, r_info in self.robots_info.items() if r_info['status'] == 'vivo']
+                if len(vivos) == 1 and self.robots_info['P']['status'] == 'morto': # Se sobrar só um e não for o jogador
+                    self.locks['game_over'].set()
+                    self.locks['winner_id'].value = vivos[0]
+                    self.logger.info(f"Jogo encerrado! Vencedor: {vivos[0]}")
+                elif len(vivos) == 1 and self.robots_info['P']['status'] == 'vivo': # Se sobrar só o jogador
+                    self.locks['game_over'].set()
+                    self.locks['winner_id'].value = 'P'
+                    self.logger.info(f"Jogo encerrado! Vencedor: P (Você!)")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)